### PR TITLE
Adds support for workspaces in conversation

### LINKED ISSUE #152: Python SDK requires Conversation API implementation for 'workspaces' endpoint.
The new Conversation endpoint supports the management of workspaces via the /workspaces API endpoint.  This functionality needs to be implemented ASAP

https://watson-api-explorer.mybluemix.net/apis/conversation-v1

### PR DESCRIPTION
**Pull request by kognate**
Conversation now has support for workspaces.
I also added a more informative jupyter notebook example
that creates a workspace with intents and dialog
for the message example.

Fixes #152

### CODE REVIEW

File: watson_developer_cloud/conversation_v1.py [hunk 1]
```diff
@@ -30,6 +30,68 @@ def __init__(self, version, url=default_url, **kwargs):
...
+        Deletes a given workspace.
+        :param: workspace_id the guid of the workspace_id
+        """
+        params = {'version': self.version}
+        return self.request(method='DELETE',
+                            url='/v1/workspaces/{0}'.format(workspace_id),
+                            params=params,
+                            accept_json=True)
+
+    def create_workspace(self, workspace_data):
```
Comment (jsstylos) [inline watson_developer_cloud/conversation_v1.py]: Usually for the Python SDK we expose all of the top-level fields of a JSON body as individual parameters.

Comment (jsstylos) [inline watson_developer_cloud/conversation_v1.py]: (Like in the `.message` method)

Comment (kognate) [inline watson_developer_cloud/conversation_v1.py]: There are a lot of them.  Much more than in the message method.  and their mostly optional.  what's the best way to handle that?

Comment (jsstylos) [inline watson_developer_cloud/conversation_v1.py]: It's, `name, description, language, metadata, counterexamples, dialog_nodes, entities, intents`? I would still make them all top-level and provide `=None` defaults for most of them.

Comment (kognate) [inline watson_developer_cloud/conversation_v1.py]: OK.

**Pull request merged**